code/controllers/subsystem/media.dm

```diff
@@ -115,16 +84,87 @@ SUBSYSTEM_DEF(media)
 		for(var/jd_tag in jd_tag_cache)
 			LAZYADD(tracks_by_tag[jd_tag], media_datum)
 
-		//For loop begins at L:32
 	return ..()
 
+/// Quarantine proc for json decoding, Has handling code for most reasonable issues with file structure, and can safely die.
+/// Returns -1/MEDIA_LOAD_FAILED on error, a list on success, and null on suicide.
+/datum/controller/subsystem/media/proc/decode_or_null(basedir,json_record)
+	PRIVATE_PROC(TRUE)
+	var/file_contents = rustg_file_read("[basedir][json_record]")
+	if(!length(file_contents))
+		log_load_fail(json_record,list("ERR_FATAL","File is empty."))
+		return MEDIA_LOAD_FAILED
+	if(!rustg_json_is_valid(file_contents))
+		log_load_fail(json_record,list("ERR_FATAL","JSON content is invalid!"))
+		return MEDIA_LOAD_FAILED
+	return json_decode(file_contents)
+
+/// Log a failure to load a specific media track, and notify admins.
+/datum/controller/subsystem/media/proc/log_load_fail(entry,list/error_tuple)
+	PRIVATE_PROC(TRUE)
+	LAZYINITLIST(errored_files)
+	errored_files[entry] = "[error_tuple[1]]:[error_tuple[2]]"
+	log_config("MEDIA: [entry] FAILED: [error_tuple[1]]:[error_tuple[2]]")
+	if(!invalid_jsons_exist)
+		//Only fire this once. Just check config_error...
+		invalid_jsons_exist = TRUE
+		spawn(0)
+			UNTIL(SSmedia.initialized)
+			message_admins("MEDIA: At least 1 Media JSON is invalid. Please check SSMedia.errored_files or config_error.log")
+	return
+
+/// Run media validation checks. Returns null on success, or a log_load_failure compatible tuple-list on failure.
+/datum/controller/subsystem/media/proc/validate_media(json_data, jd_full_filepath, jd_tag_cache)
+	if(!json_data || !jd_full_filepath || !jd_tag_cache)
+		stack_trace("BAD CALLING ARGUMENTS TO VALIDATE_MEDIA")
+		return list("ERR_FATAL", "Record validation was called with bad arguments: [json_data || "#FALSY_DATA"], [jd_full_filepath || "#FALSY_DATA"], [english_list(jd_tag_cache, nothing_text = "#FALSY_DATA")]")
```

Comment:
can you use more to the point errors, errors are in english we arent writing an operating system

Reply:
You should never get this particular error *EVER* unless something is so bad you *should* be getting a developer.

If you have a problem with any of the more general errors, let me know.